packages/snap-preact-components/src/components/Organisms/Recommendation/Recommendation.tsx

```diff
@@ -72,12 +73,12 @@ export const Recommendation = observer((properties: RecommendationProps): JSX.El
 		throw new Error(`<Recommendation> Component requires 'controller' prop with an instance of RecommendationController`);
 	}
 
-	if (children && children.length !== controller.store.results.length) {
+	if (!results && children && children.length !== controller.store.results.length) {
```

Comment:
There is a case where you may want to pass both `children` and `results` - in this case would we still want to ensure that children and results are of the same length?

Perhaps we could just move this below the `resultsToRender` declaration and compare `children.length` to `resultsToRender.length`?